rfcs/20190722-tflite-training.md

```diff
@@ -0,0 +1,427 @@
+# On-Device Training with TensorFlow Lite
+
+Status        | Proposed
+:------------ | :---------------------------------------------------------------
+**Author(s)** | Yu-Cheng Ling (ycling@google.com)
+**Sponsor**   | Andrew Selle (aselle@google.com), Jared Duke (jdduke@google.com)
+**Updated**   | 2019-07-22
+
+## Overview & Roadmap
+
+TensorFlow Lite is TensorFlow's recommended solution for on-device machine
+learning. Initially the project focuses on inference, but more and more users
+are asking for on-device training recently.
+
+The doc scopes a multi-quarter effort to get generalized & optimized on-device
+training working with TensorFlow Lite. The project can be broken down into a few
+milestones:
+
+**Milestone 1: Working prototype for basic training (e.g. fully connected / conv layers only)**<br/>
+Goal: Have a working prototype to train over fully connected & convolutional layers.
+
+**Milestone 2A: Optimized basic training**<br/>
+Goal: Make inference & training performance comparable with TFMobile.
+
+**Milestone 2B: Enable on-device training loop with TensorFlow functionality**<br/>
+Goal: Encode the on-device training loop inside a TensorFlow model, with TensorFlow functionality like tf.Example and tf.DataSet.
+
```

Comment:
What is in scope here? Eg... Preprocessing? Control flow? There is a whole support matrix here to consider-- how much is covered?

Reply:
Control flow is definitely in the roadmap. 

Also, be able to leverage the Flex runtime to use TensorFlow kernels for TF Example and DataSet etc. This for users who are requesting to port the TFMobile flow as is (but using TF Example and DataSet isn't necessarily optimized)